tests/System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryTests.cs

```diff
@@ -1244,6 +1244,26 @@ public void MockDirectory_GetDirectories_AbsoluteWithNoSubDirectories_ShouldRetu
             Assert.That(actualResult, Is.Empty);
         }
 
+        [TestCase(@"Folder\SubFolder")]
```

Comment:
I think we're still missing some test cases here. I just played around with a tiny test program:

```csharp

var directories = Directory.GetDirectories(args[0], args[1]);
foreach (var directory in directories)
{
    Console.WriteLine(directory);
}
```

Here are some scenarios that I guess don't work with the current approach:

![image](https://user-images.githubusercontent.com/1540469/158884210-2b9a1cb3-e881-4b6b-8e05-79405079ba51.png)

Reply:
I've added the test cases but in the test about relative paths above